### ISSUE TITLE
flatpak run fails if host filesystem is not accessible

### ISSUE THREAD

**Issue by pyamsoft** [jlHertel/adriconf#65]
Apologies if flatpak specific issues should not go here.

When running `adriconf` in an X11 flatpak which has no access to the host filesystem `!host`, the application fails to start with

```
INFO: adriconf running on X11
ERROR: Generic Error: boost::filesystem::directory_iterator::construct: No such file or directory: "/var/run/host/usr/share/drirc.d/"
```
As of commit: `22991de80fcc618fdf9556390fe39e6d480d220afffd3ddf5a7000ec8054dc86`

Because the flatpak has no access to the host system, the directory `/var/run/host/usr` will not be created.

Sadly, I have not looked at the source code, and while I understand the application does not expect to run in these conditions, would it be possible to ignore the fact that this directory does not exist and open the application anyway using any other existing directories such as the users `~/.drirc`, or is the access to the host filesystem required in this case?

**Comment by jlHertel** [comment]
When mesa loads, it will load files from various locations to determine what is the configuration for the current application.
The directory `/usr/share/drirc.d/` is the place used by mesa to store the default configuration files. It can happen that the directory is empty and this is totally fine, but denying access is something that will not make sense.

Is there any reason why you want to remove this access?
Maybe we can change something to make our current access more restricted. I'm open to discuss options here, as long as we can continue to access the directory under `/usr/share/drirc.d/` and the file under  `/etc/drirc`.

**Comment by jlHertel** [comment]
Small note for myself: We should not fail in those cases, as old mesa versions will not have this directory. So we need to make loading this directory optional.

**Comment by pyamsoft** [comment]
The only reason I was running without host access was because it was part of my setup in the global flatpak overrides file.

To me the ideal solution would be for flatpak to allow filesystem access to individual files in the host /usr instead of it's current all-or-nothing approach. 
This may be more of a limitation on flatpak itself, which to my understanding does not let me bind a single folder into the /var/run/host directory. 

Ultimately I do not believe this to be a problem for the adriconf project to solve.

**Comment by pyamsoft** [comment]
Feel free to close this issue as you see fit.

Thank you for your work on the project.

**Comment by jlHertel** [comment]
I think is possible to isolate the access to only the needed files/dirs.
This gives more details on how to do it: http://docs.flatpak.org/en/latest/sandbox-permissions.html#filesystem-access

I will play around with it when I have time and try to improve the current scenario.

**Comment by jlHertel** [comment]
The issue of forcing the existence of a system-wide configuration dir has been fixed with b902653e8efa15e6c47fafa4e578759824694cec.
Please take note that adriconf will produce wrong results in case this directory is not accessible, so this commit is more to avoid hard crashes.
